pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go

```diff
@@ -100,10 +100,17 @@ func getSecurityProfileCiphers(profile *configv1.TLSSecurityProfile) (string, []
 	}
 
 	// nothing found / custom type set but no actual custom spec
+	defaultProfileSpec := configv1.TLSProfiles[configv1.TLSProfileIntermediateType]
 	if profileSpec == nil {
-		profileSpec = configv1.TLSProfiles[configv1.TLSProfileIntermediateType]
+		profileSpec = defaultProfileSpec
+	}
+
+	// If a custom spec was provided with no MinTLSVersion, use the default
+	minTLSVersion := profileSpec.MinTLSVersion
+	if minTLSVersion == "" {
```

Comment:
I think that the best place for defaulting and validation is in the API server but we would have to add it in a backward compatible way.

Reply:
Thanks for checking the cipher suites. I have just realised that the API server already validates the tls profile, here: https://github.com/openshift/kubernetes/blob/master/openshift-kube-apiserver/admission/customresourcevalidation/apiserver/validate_apiserver.go#L254

It looks like the `minTLSVersion` can be empty. I suspect this is because the std library will apply defaults depending on golang version and whether it is a client or a server. 

I would like to avoid changing existing behaviour. How about updating your controller so that it can work with empty `minTLSVersion` ? If you need a specific version then you can apply one or just call `libgocrypto.TLSVersion()` function.

Reply:
Yeah that makes sense, I pushed a new version that only affects storage controllers. Thanks for the feedback!